lib/bundle/util/key-generator.js

```diff
@@ -37,8 +37,12 @@ KeyGenerator.appendSlash = function (str) {
 };
 
 function KeyGenerator (root) {
-  this.root = root;
-  this._parsedRoot = safePointerToPath(root);
+  let rootFn = root;
+  if (!root || typeof root === "string") {
+    rootFn = () => root;
+  }
+
+  this.root = rootFn;
```

Comment:
```suggestion
  if (!root || typeof root === "string") {
    this.root = () => root;
  } else {
    this.root = root;
  }
```
or we could even have a one-liner
```suggestion
  this.root = typeof root === "function" ? root : () => root;
```